java/flight/flight-jdbc/src/main/java/org/apache/arrow/jdbc/FlightResultSet.java

```diff
@@ -0,0 +1,1062 @@
+  }
+
+  @Override
+  public boolean getBoolean(int i) throws SQLException {
+    return ResultSetHelper.getBoolean(getObject(i));
+  }
+
+  @Override
+  public boolean wasNull() throws SQLException {
+    return wasNull;
+  }
+
+  @Override
+  public void close() throws SQLException {
+    try {
+      stream.close();
+    } catch (Exception e) {
+      throw new SQLException(e);
+    }
+  }
+
+  @Override
+  public byte getByte(int i) throws SQLException {
+    return ResultSetHelper.getByte(getObject(i));
+  }
+
+  @Override
+  public short getShort(int i) throws SQLException {
+    return ResultSetHelper.getShort(getObject(i));
+  }
+
+  @Override
+  public int getInt(int i) throws SQLException {
+    return ResultSetHelper.getInt(getObject(i));
+  }
+
+  @Override
+  public long getLong(int i) throws SQLException {
+    return ResultSetHelper.getLong(getObject(i));
+  }
+
+  @Override
+  public float getFloat(int i) throws SQLException {
+    return ResultSetHelper.getFloat(getObject(i));
+  }
+
+  @Override
+  public double getDouble(int i) throws SQLException {
+    return ResultSetHelper.getDouble(getObject(i));
+  }
+
+  @Deprecated
+  @Override
+  public BigDecimal getBigDecimal(int i, int scale) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public byte[] getBytes(int i) throws SQLException {
+    return ResultSetHelper.getBytes(getObject(i));
+  }
+
+  @Override
+  public Date getDate(int i) throws SQLException {
+    return ResultSetHelper.getDate(getObject(i));
+  }
+
+  @Override
+  public Time getTime(int i) throws SQLException {
+    return ResultSetHelper.getTime(getObject(i));
+  }
+
+  @Override
+  public Timestamp getTimestamp(int i) throws SQLException {
+    return ResultSetHelper.getTimestamp(getObject(i));
+  }
+
+  @Override
+  public InputStream getAsciiStream(int i) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public InputStream getUnicodeStream(int i) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public InputStream getBinaryStream(int i) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public String getString(String columnName) throws SQLException {
+    return ResultSetHelper.getString(getObject(columnName));
+  }
+
+  @Override
+  public boolean getBoolean(String columnName) throws SQLException {
+    return ResultSetHelper.getBoolean(getObject(columnName));
+  }
+
+  @Override
+  public byte getByte(String columnName) throws SQLException {
+    return ResultSetHelper.getByte(getObject(columnName));
+  }
+
+  @Override
+  public short getShort(String columnName) throws SQLException {
+    return ResultSetHelper.getShort(getObject(columnName));
+  }
+
+  @Override
+  public int getInt(String columnName) throws SQLException {
+    return ResultSetHelper.getInt(getObject(columnName));
```

Comment:
There are boxing/unboxing overheads here, which should be eliminated in the future.